### PR TITLE
Model for BusinessSubscriptionDetails DES API 1566

### PR DESCRIPTION
**Pull request by chandraAndem**

### CODE REVIEW

File: app/models/individual/subscription/MultipleIncomeSourcesSubscriptionResponse.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import models.ConnectorError
+import play.api.libs.json.{Json, OFormat}
+
+case class IncomeSourceId(incomeSourceId: String)
+
+object IncomeSourceId {
+  implicit val format: OFormat[IncomeSourceId] = Json.format[IncomeSourceId]
+}
+
+case class MultipleIncomeSourcesSubscriptionSuccess(incomeSourceIds: List[IncomeSourceId])
```
Comment (jameshforster) [inline app/models/individual/subscription/MultipleIncomeSourcesSubscriptionResponse.scala:28]: Same naming issues as above.

**Review by jameshforster — CHANGES_REQUESTED** [review]

File: app/connectors/individual/subscription/httpparsers/MultipleIncomeSourcesSubscriptionResponseHttpParser.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+package connectors.individual.subscription.httpparsers
+
+import models.individual.subscription._
+import play.api.http.Status.OK
+import play.api.libs.json.JsSuccess
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+object MultipleIncomeSourcesSubscriptionResponseHttpParser {
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/connectors/individual/subscription/httpparsers/MultipleIncomeSourcesSubscriptionResponseHttpParser.scala:24]: Make sense, Done.

File: app/models/individual/subscription/MultipleIncomeSourcesSubscriptionResponse.scala (hunk 1 above)
Comment (chandraAndem) [reply app/models/individual/subscription/MultipleIncomeSourcesSubscriptionResponse.scala:28]: Done

File: app/connectors/individual/subscription/MultipleIncomeSourcesSubscriptionConnector.scala [hunk 3]
```diff
@@ -30,14 +32,19 @@ class MultipleIncomeSourcesSubscriptionConnector @Inject()(val appConfig: AppCon
...
 
+  def subscribe(nino: String, request: BusinessSubscriptionDetailsModel)(implicit hc: HeaderCarrier): Future[PostMultipleIncomeSourceSubscriptionResponse] =
+    http.POST[BusinessSubscriptionDetailsModel, PostMultipleIncomeSourceSubscriptionResponse](subscribeUrl(nino), request)
+
 }
 
 object MultipleIncomeSourcesSubscriptionConnector {
 
   def signUpUri(nino: String): String = "/mis/sign-up/" + nino
+  def subscribeUri(nino: String): String = "/mis/subscribe/" + nino
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/connectors/individual/subscription/MultipleIncomeSourcesSubscriptionConnector.scala:48]: Done

File: app/connectors/individual/subscription/MultipleIncomeSourcesSubscriptionConnector.scala [hunk 4]
```diff
@@ -30,14 +32,19 @@ class MultipleIncomeSourcesSubscriptionConnector @Inject()(val appConfig: AppCon
                                                           (implicit ec: ExecutionContext) {
 
   def signUpUrl(nino: String): String = appConfig.subscriptionUrl + MultipleIncomeSourcesSubscriptionConnector.signUpUri(nino)
+  def subscribeUrl(nino: String): String = appConfig.subscriptionUrl + MultipleIncomeSourcesSubscriptionConnector.subscribeUri(nino)
 
   def signUp(nino: String)(implicit hc: HeaderCarrier): Future[PostMultipleIncomeSourceSignUpResponse] =
     http.POSTEmpty[PostMultipleIncomeSourceSignUpResponse](signUpUrl(nino))
 
+  def subscribe(nino: String, request: BusinessSubscriptionDetailsModel)(implicit hc: HeaderCarrier): Future[PostMultipleIncomeSourceSubscriptionResponse] =
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/connectors/individual/subscription/MultipleIncomeSourcesSubscriptionConnector.scala:40]: Done

File: app/connectors/individual/subscription/MultipleIncomeSourcesSubscriptionConnector.scala [hunk 5]
```diff
@@ -30,14 +32,19 @@ class MultipleIncomeSourcesSubscriptionConnector @Inject()(val appConfig: AppCon
                                                           (implicit ec: ExecutionContext) {
 
   def signUpUrl(nino: String): String = appConfig.subscriptionUrl + MultipleIncomeSourcesSubscriptionConnector.signUpUri(nino)
+  def subscribeUrl(nino: String): String = appConfig.subscriptionUrl + MultipleIncomeSourcesSubscriptionConnector.subscribeUri(nino)
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/connectors/individual/subscription/MultipleIncomeSourcesSubscriptionConnector.scala:35]: Done

**Pull request merged**